xaitk_saliency/impls/gen_detection_sal/DRISE.py

```diff
@@ -0,0 +1,124 @@
+    different object classes).
+    """
+    def __init__(
+        self,
+        proximity_metric: str = 'cosine'
+    ):
+
+        try:
+            # Attempting to use chosen comparision metric
+            cdist([[1], [1]], [[1], [1]], proximity_metric)
+            self.proximity_metric: str = proximity_metric
+        except ValueError:
+            raise ValueError("Chosen comparision metric not supported or",
+                             "may not be available in scipy")
+
+    def intersect(self, box_a, box_b):
+        """ We resize both tensors to [A,B,2] without new malloc:
+        [A,2] -> [A,1,2] -> [A,B,2]
+        [B,2] -> [1,B,2] -> [A,B,2]
+        Then we compute the area of intersect between box_a and box_b.
+        Args:
+          box_a: (tensor) bounding boxes, Shape: [A,4].
+          box_b: (tensor) bounding boxes, Shape: [B,4].
+        Return:
+          (tensor) intersection area, Shape: [A,B].
+        """
+        A = box_a.size(0)
+        B = box_b.size(0)
+        min_xy = torch.min(box_a[:, 2:].unsqueeze(1).expand(A, B, 2),
+                           box_b[:, 2:].unsqueeze(0).expand(A, B, 2))
+        max_xy = torch.max(box_a[:, :2].unsqueeze(1).expand(A, B, 2),
+                           box_b[:, :2].unsqueeze(0).expand(A, B, 2))
+        inter = torch.clamp((min_xy - max_xy + 1 ), min=0)
+        return inter[:, :, 0] * inter[:, :, 1]
+
+    def jaccard(self, box_a, box_b):
+        """Compute the jaccard overlap of two sets of boxes.  The jaccard overlap
+        is simply the intersection over union of two boxes.  Here we operate on
+        ground truth boxes and default boxes.
+        E.g.:
+            A ∩ B / A ∪ B = A ∩ B / (area(A) + area(B) - A ∩ B)
+        Args:
+            box_a: (tensor) Ground truth bounding boxes, Shape: [num_objects,4]
+            box_b: (tensor) Prior boxes from priorbox layers, Shape: [num_priors,4]
+        Return:
+            jaccard overlap: (tensor) Shape: [box_a.size(0), box_b.size(0)]
+        """
+        inter = self.intersect(box_a, box_b)
+        area_a = ((box_a[:, 2]-box_a[:, 0]) *
+                  (box_a[:, 3]-box_a[:, 1])).unsqueeze(1).expand_as(inter)
+        area_b = ((box_b[:, 2]-box_b[:, 0]) *
+                  (box_b[:, 3]-box_b[:, 1])).unsqueeze(0).expand_as(inter)
+        union = area_a + area_b - inter
+
+        return inter / union
+
+    def generate(
+        self,
+        ref_dets: np.ndarray,
+        perturbed_dets: np.ndarray,
+        perturb_masks: np.ndarray,
+    ) -> np.ndarray:
+
+        if len(perturb_masks) != len(perturbed_dets):
+            raise ValueError("Number of perturbation masks and respective",
+                 "detections vector do not match.")
+
+        self.n_targets = len(ref_dets)
+        n_targets = len(ref_dets)
+        self.N = len(perturb_masks)
+        self.input_size = perturb_masks[0].shape
+        self.masks = torch.from_numpy(perturb_masks)
```

Comment:
Does this need to use torch for these operations? While this is an implementation an may require optional dependencies, it seems that this is just performing array operations that should have numpy equivalents. If so, then we should reduce the torch dependency down to numpy (which is already required).